src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react';
+import { TodoItem } from './TodoItem';
+import { Todo } from '../types/Todo';
+
+export const TodoInfo = React.memo(
+  ({
+    todosFromServer,
+    askTodos,
+    setErrorMessage,
+    setBlock,
+  }: {
+    todosFromServer: Todo[] | undefined;
+    askTodos: (url: string) => void;
+    setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+    setBlock: {
+      setCountComplited: React.Dispatch<React.SetStateAction<boolean>>;
+      setCountNotComplited: React.Dispatch<React.SetStateAction<boolean>>;
+    }
+  }) => {
+    useEffect(() => {
+      if (todosFromServer) {
+        const isCount = todosFromServer.some(
+          (todo: Todo) => todo.completed === true,
+        );
+        const isNotCount = todosFromServer.some(
+          (todo: Todo) => todo.completed === false,
+        );
+
+        setBlock.setCountNotComplited(isNotCount);
+        setBlock.setCountComplited(isCount);
```

Comment:
it's very bad approach to set State from child component using the object. it's not obvious.
maybe try to use `useMemo` 
```
const countNotComplited = useMemo(() => todosFromServer.some(
  (todo: Todo) => todo.completed === false,
), [todosFromServer]);
```